tools/ZydisFuzzShared.c

```diff
@@ -147,6 +147,67 @@ void ZydisValidateEnumRanges(const ZydisDecodedInstruction* insn,
+    {                                                                                              \
+        fprintf(stderr, "Value " #value " = 0x%016" PRIX64 " does not match any of the expected"   \
+            " values: " #a " = 0x%016" PRIX64 ", " #b " = 0x%016" PRIX64 ", "                      \
+            #c " = 0x%016" PRIX64 "\n",                                                            \
+            (ZyanU64)(value), (ZyanU64)(a), (ZyanU64)(b), (ZyanU64)(c));                           \
+        abort();                                                                                   \
+    }
+
+
+#   define ZYDIS_CHECK_VALUE_ONE_OF_4(value, a, b, c, d)                                           \
+    if ((ZyanU64)(value) != (ZyanU64)(a) && (ZyanU64)(value) != (ZyanU64)(b) &&                    \
+        (ZyanU64)(value) != (ZyanU64)(c) && (ZyanU64)(value) != (ZyanU64)(d))                      \
+    {                                                                                              \
+        fprintf(stderr, "Value " #value " = 0x%016" PRIX64 " does not match any of the expected "  \
+            "values. " #a " = 0x%016" PRIX64 ", " #b " = 0x%016" PRIX64 ", " #c                    \
+            " = 0x%016" PRIX64 ", " #d " = 0x%016" PRIX64 "\n",                                    \
+            (ZyanU64)(value), (ZyanU64)(a), (ZyanU64)(b), (ZyanU64)(c), (ZyanU64)(d));             \
+        abort();                                                                                   \
+    }
+
+#   define ZYDIS_CHECK_VALUE_ONE_OF_5(value, a, b, c, d, e)                                        \
+    if ((ZyanU64)(value) != (ZyanU64)(a) && (ZyanU64)(value) != (ZyanU64)(b) &&                    \
+        (ZyanU64)(value) != (ZyanU64)(c) && (ZyanU64)(value) != (ZyanU64)(d) &&                    \
+        (ZyanU64)(value) != (ZyanU64)(e))                                                          \
+    {                                                                                              \
+        fprintf(stderr, "Value " #value " = 0x%016" PRIX64 " does not match any of the expected "  \
+            "values. " #a " = 0x%016" PRIX64 ", " #b " = 0x%016" PRIX64 ", " #c                    \
+            " = 0x%016" PRIX64 ", " #d " = 0x%016" PRIX64 ", " #e " = 0x%016" PRIX64 "\n",         \
+            (ZyanU64)(value), (ZyanU64)(a), (ZyanU64)(b), (ZyanU64)(c), (ZyanU64)(d),              \
+            (ZyanU64)(e));                                                                         \
+        abort();                                                                                   \
+    }
+
+
+
+#define ZYDIS_CHECK_VALUE_ONE_OF_N(value, ...) \
+        GET_MACRO(__VA_ARGS__, ZYDIS_CHECK_VALUE_ONE_OF_5, ZYDIS_CHECK_VALUE_ONE_OF_4, ZYDIS_CHECK_VALUE_ONE_OF_3, ZYDIS_CHECK_VALUE_ONE_OF_2, ZYDIS_CHECK_VALUE_ONE_OF_1)(value, __VA_ARGS__)
+
+#define GET_MACRO(_1, _2, _3, _4, _5, NAME, ...) NAME
+
```

Comment:
I'm not a fan of these macros but this was the only solution i could come up with.

Reply:
Thanks for taking the effort to create a generic implementation. For me personally, I would have been fine with just a manual check without a macro as well. I let @mappzor decide about this, as it's mostly the codebase he worked on.

Reply:
I would get rid of macros for this one. A small utility function that checks against known goods (0, 8, 16, 32, 64) would do the trick. It doesn't even have to accept array of good values as parameter, they could be hardcoded as static array within the function. If we ever need more flexiblity we'll just update the function when it's actually needed.